### ISSUE TITLE
Bug: Selection formatting/styling is lost when you indent a list

### ISSUE THREAD

**Issue by gigantedocil** [facebook/lexical#6476]
If you create a list item with some style, press enter to create a new item at the same level and then press tab you lose the formatting on the selection.

Lexical version: latest

## Steps To Reproduce

1. Create a list that has some styling, for example bold:

![image](https://github.com/user-attachments/assets/23b11d8c-0672-41fa-b70b-18fcf02b8a9b)

2. Press enter to create a new list item of the same level (bold styling remains on selection.

![image](https://github.com/user-attachments/assets/6e37ecf1-bd10-428e-a3c1-298c2ff66eee)

3. Press tab to indent the list item (bold styling disappears).

![image](https://github.com/user-attachments/assets/589e9e98-6799-4e66-8287-ba8af9810887)

Link to code example:

Can easily be reproduced on the playground.

## The current behavior

Selection formatting is lost after indenting a list item.

## The expected behavior

Selection formatting is not lost after indenting a list item.

## Impact of fix

Currently an important bug on my app is caused by this.

**Comment by citruscai** [comment]
i will work on this 😎

**Comment by gigantedocil** [comment]
@citruscai, awesome, thanks! Were you able to reproduce it?

**Comment by citruscai** [comment]
> @citruscai, awesome, thanks! Were you able to reproduce it?

yes! i was able to reproduce it in the playground and saw the issue. i also reproduced it in google docs and google docs does keep the original format after you indent a list item. so i am happy that i found an issue and excited to tackle it 😄

**Comment by citruscai** [comment]
> > @citruscai, awesome, thanks! Were you able to reproduce it?
> 
>  @gigantedocil  yes! i was able to reproduce it in the playground and saw the issue. i also reproduced it in google docs and google docs does keep the original format after you indent a list item. so i am happy that i found an issue and excited to tackle it 😄

**Comment by gigantedocil** [comment]
@citruscai, amazing work! Thanks!

**Comment by citruscai** [comment]
@gigantedocil hi!! did this fix the issue in your project?
